packages/blocks/src/components/database-modal/database-modal.ts

```diff
@@ -0,0 +1,163 @@
+import { getCurrentBlockRange, getDefaultPage } from '@blocksuite/blocks';
+import {
+  DatabaseKanbanViewIcon,
+  DatabaseSearchClose,
+  DatabaseTableViewIcon,
+  MoreHorizontalIcon,
+} from '@blocksuite/global/config';
+import { assertExists } from '@blocksuite/global/utils';
+import type { Page } from '@blocksuite/store';
+import { html, LitElement, type TemplateResult } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+
+import { styles } from './styles.js';
+
+type DatabaseViewName = 'table' | 'kanban';
+
+type DatabaseView = {
+  type: DatabaseViewName;
+  text: string;
+  icon: TemplateResult;
+  description?: string;
+  isComingSoon?: boolean;
+};
+
+const databaseViews: DatabaseView[] = [
+  {
+    type: 'table',
+    text: 'Table view',
+    icon: DatabaseTableViewIcon,
+  },
+  {
+    type: 'kanban',
+    text: 'Kanban view',
+    icon: DatabaseKanbanViewIcon,
+    description: 'Coming soon',
+    isComingSoon: true,
+  },
+];
+
+@customElement('affine-database-modal')
+export class DatabaseModal extends LitElement {
+  static styles = styles;
+
+  @property()
+  page!: Page;
+
+  @state()
+  private _selectedView: DatabaseViewName = 'table';
+
+  private _convertToDatabase(viewType: DatabaseViewName) {
+    this._hide();
+    this.page.captureSync();
+
+    const range = getCurrentBlockRange(this.page);
+    assertExists(range);
+    const models = range.models;
+
+    const parentModel = this.page.getParent(models[0]);
+    assertExists(parentModel);
+
+    // default column
+    const tagColumnId = this.page.db.updateColumn({
+      name: 'Tag',
+      type: 'multi-select',
+      width: 200,
+      hide: false,
+      selection: [],
+    });
+
+    const id = this.page.addBlock(
+      'affine:database',
+      {
+        columns: [tagColumnId],
+        titleColumnName: 'Title',
+      },
+      parentModel,
+      parentModel.children.indexOf(models[0])
+    );
+
+    const databaseModel = this.page.getBlockById(id);
+    assertExists(databaseModel);
+    this.page.moveBlocks(models, databaseModel);
+
+    // Try clean block selection
+    const defaultPageBlock = getDefaultPage(this.page);
+    assertExists(defaultPageBlock);
+    if (!defaultPageBlock.selection) {
+      // In the edgeless mode
+      return;
+    }
+    defaultPageBlock.selection.clear();
+  }
+
+  private _hide() {
+    this.dispatchEvent(
+      new CustomEvent<DatabaseDetail>('hide', {
+        detail: { type: 'hide' },
+      })
+    );
```

Comment:
We have phased out the event style because it is not type-friendly and difficult to trace calls. Could you modify it to trigger `abortController.abort` instead? I apologize for any confusion the old code may have caused.

```suggestion
    this.abortController.abort();
```

Reply:
get~